### ISSUE TITLE
Wire resource cleanup error

### ISSUE THREAD

**Issue by lburgazzoli** [OpenHFT/Chronicle-Wire#22]
It appears that ref-counters are broken after a wire is applied to a mapped bytes:

```java
java.lang.IllegalStateException: Released

	at net.openhft.chronicle.core.ReferenceCounter.release(ReferenceCounter.java:47)
	at net.openhft.chronicle.bytes.AbstractBytes.release(AbstractBytes.java:329)
	at net.openhft.chronicle.wire.WireResourcesTest.testMappedBytesWireRelease(WireResourcesTest.java:47)
```

Test case here: https://github.com/OpenHFT/Chronicle-Wire/pull/21

**Comment by peter-lawrey** [comment]
Thank you.

I have extended the test and made the behaviour more consistent.

https://github.com/OpenHFT/Chronicle-Wire/blob/master/src/test/java/net/openhft/chronicle/wire/WireResourcesTest.java

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.101](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
